home-exercises/src/main/java/advancedtest/exercise3/Joiner.java

```diff
@@ -0,0 +1,23 @@
+package advancedtest.exercise3;
+
+import java.util.function.BiFunction;
+
+public class Joiner<T> {
+    private final String seperator;
+    private final T input;
+    private final BiFunction<T, String, String> join;
+
+    public Joiner( T input, String seperator, BiFunction<T, String, String> join) {
```

Comment:
Uzdevumā bija konkrēti minēts, kas jāsaņem konstruktorā :)
"Design the Joiner<T> class, which in the constructor will take a separator (string)"

Par mainīgu daudzumu - varēja gan ar List gan ar varargs
public String joinTogether(T... items)

Reply:
Bija piemirsies pats varargs un saputrojos, kuru funkcijas tipu izmantot. Visu izlaboju un izpildiju